keras_nlp/applications/bert.py

```diff
@@ -66,118 +67,118 @@ class BertModel(keras.Model):
     def __init__(
         self,
         vocab_size,
-        num_layers=12,
-        hidden_size=768,
-        dropout=0.1,
-        num_attention_heads=12,
-        inner_size=3072,
+        num_layers,
+        hidden_size,
+        num_attention_heads,
+        inner_size,
         inner_activation="gelu",
         initializer_range=0.02,
+        dropout=0.1,
         max_sequence_length=512,
         type_vocab_size=2,
-        **kwargs,
+        **kwargs
     ):
-        super().__init__(**kwargs)
 
-        self.vocab_size = vocab_size
-        self.hidden_size = hidden_size
-        self.num_layers = num_layers
-        self.num_attention_heads = num_attention_heads
-        self.max_sequence_length = max_sequence_length
-        self.type_vocab_size = type_vocab_size
-        self.inner_size = inner_size
-        self.inner_activation = keras.activations.get(inner_activation)
-        self.initializer_range = initializer_range
-        self.initializer = keras.initializers.TruncatedNormal(
+        # Create lambda functions from input params
+        inner_activation_fn = keras.activations.get(inner_activation)
+        initializer_fn = keras.initializers.TruncatedNormal(
             stddev=initializer_range
         )
-        self.dropout = dropout
 
-        self._embedding_layer = keras.layers.Embedding(
-            input_dim=vocab_size,
-            output_dim=hidden_size,
-            embeddings_initializer=self.initializer,
-            name="word_embeddings",
+        # Functional version of model
+        token_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="input_ids"
+        )
+        segment_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="segment_ids"
+        )
+        # TODO(jbischof): improve handling of masking following
+        # https://www.tensorflow.org/guide/keras/masking_and_padding
+        input_mask = keras.Input(
+            shape=(None,), dtype="int32", name="input_mask"
         )
 
-        self._position_embedding_layer = keras_nlp.layers.PositionEmbedding(
-            initializer=self.initializer,
+        # Embed tokens, positions, and segment ids.
+        token_embedding = keras.layers.Embedding(
+            input_dim=vocab_size,
+            output_dim=hidden_size,
+            name=TOKEN_EMBEDDING_LAYER_NAME,
+        )(token_id_input)
+        position_embedding = keras_nlp.layers.PositionEmbedding(
```

Comment:
I think we are relying on the `__init__` files for the imports here. This is safe to do as a client of the library (e.g. examples dir), but from within the library it is probably better practice to direct import each of these from their files.

Reply:
Done! The tiny awkward is that we now have `keras.layers.X` if it's core Keras and `X` if it's `keras-nlp.layers`, but not the end of the world.

Reply:
Yeah. This is a good point. If we want the application code to be copy pasteable, using the exported library imports might be a good call here.

Could we just use the exported locations for all intra-library imports for keras_nlp? Or are would we run into some circular import shenanigans if we do that. It would be slightly annoying to have a separate style of imports just for `keras_nlp.models`, but maybe it is worth it for copy-paste-ability.